### PR TITLE
pkg/cluster: add unit tests and split options into separate Go files

### PR DESCRIPTION
**Pull request by dcbw**
@shettyg @squeed @danwinship @rajatchopra

### CODE REVIEW

File: go-controller/pkg/util/gateway-init.go [hunk 1]
```diff
@@ -136,18 +135,16 @@ func GatewayInit(clusterIPSubnet []string, nodeName, nicIP, physicalInterface,
 		gatewayRouter, "--", "set", "logical_router", gatewayRouter,
 		"options:chassis="+systemID, "external_ids:physical_ip="+physicalIP)
 	if err != nil {
-		logrus.Errorf("Failed to create logical router %v, stdout: %q, "+
+		return fmt.Errorf("Failed to create logical router %v, stdout: %q, "+
```
Comment (squeed) [inline go-controller/pkg/util/gateway-init.go]: Can we use `pkg.errors/Errwrap` here? it's the new idiom.

Comment (dcbw) [reply go-controller/pkg/util/gateway-init.go]: @squeed yeah, could do, but I'd rather as a separate follow-up commit to reduce the actual diff

**Comment by shettyg** [comment]
I see a "asdfasdf" commit.  I guess, it was not the intention?

**Comment by dcbw** [comment]
> I see a "asdfasdf" commit. I guess, it was not the intention?

@shettyg it was to kick Travis to get the new testcases to work with Sudo.  But for some reason re-pushes to this issue and https://github.com/openvswitch/ovn-kubernetes/pull/469 weren't getting Travis CI runs.  I try rebasing to fix the new conflict and repushing.

**Comment by dcbw** [comment]
@shettyg @rajatchopra @squeed @danwinship ready for another review, PTAL thanks!

**Comment by shettyg** [comment]
Looks like a lot of moving code around. Did you try this out in some way?

**Comment by shettyg** [comment]
I did a round of sanity testing with vagrant and looks sane to me. So /lgtm

**Comment by dcbw** [comment]
> Looks like a lot of moving code around. Did you try this out in some way?

@shettyg I ran it in a few simple configurations, but the unit tests were added before the reorg so we could be 100% confident that I wasn't changing any of the command ordering.

**Pull request merged**